### ISSUE TITLE
Changing Zutilo Preferences Doesn't Affect Interface

### ISSUE THREAD

**Issue by AlexanderSavenkov** [wshanks/Zutilo#232]
I'm trying to get additional menu items into Zutilo submenu (or Zotero context menu) via Tools → Zutilo Preferences → User interface.

No matter which adjustments I make, after pressing OK there are only four default menu items in Zutilo submenu and there are no additional items in Zotero context menu.

Restarting Zotero doesn't help. All the updates are there as well. Ready to provide additional info if required.

OS: Windows 10 Pro 21H2 19044.1826
Zotero: 6.0.9
Zutilo: 3.10.0

**Comment by qqobb** [comment]
I'm not sure what the issue could be. You could try these steps:

- Disable all other plugins in "Tools" -> "Add-ons".

- Change a preference in "Tools" -> "Zutilo Preferences...". Then check if it updates as expected in Zotero's [Config Editor](https://www.zotero.org/support/preferences/advanced#general).

- Reset all Zutilo preferences to their default values. (Right-click prefs in the Config Editor.)

- Check if you get Zutilo-related errors. Disable Zutilo in "Tools" -> "Add-ons". Open "Tools" -> "Developer" -> "Error Console", clear everything. Enable Zutilo and try to use some of its functions. (If you see a relevant message in the Error Console, you can select it and copy it with Ctrl+C.) There's also "Help" -> "Debug Output Logging". Enable it and select "View Output".

**Comment by wobintosh** [comment]
I think I might have the same issue.

**Expected behavior**
I want to use "Paste into empty item fields" and "Paste non-empty item fields" from the Zutilo's context menu. If I enable only those two, there is no Zutilo Context-Menu at all.

**Actual behavior**
To narrow down the problem I selected all options from "Zotero item menu" to be shown in Zutilo's context menu. As expected from my previous experiments, not all are shown in the context menu. The missing options are:

- Paste into empty item fields
- Paste non-empty item fields
- Paste all item fields
- Paste item type

Tested a) in a saved search with only magazine articles and b) regular collections with only magazine articles and mixed articles/books.

![2022-09-05 14_27_37-Zutilo Preferences](https://user-images.githubusercontent.com/15612193/188449855-b135e22c-b8ee-4729-8126-c3be6a0efcb7.png)
![2022-09-05 14_28_09-Zutilo context menu](https://user-images.githubusercontent.com/15612193/188449858-e2d5737e-073f-46d8-a7f4-026f3c380166.png)

- No other Plugins enabled
- Changed Preferences update in Config Editor
- Reset Preferences to default value doesn't change the behaviour (the expected behaviour naturally requires me to change the settings)
- After uninstalling, re-installing and testing there are only two Zutilo-related errors in the console: "1662380335210	addons.xpi	WARN	Attempted to remove zutilo@www.wesailatdawn.com from app-profile but it was already gone" and "No chrome package registered for chrome://zutilo/skin/zutilo.png" - which do not seem to be related to my problem.

**Versions**
Windows 10 Pro Version 2004 (Build 19041.1415)
Zotero 6.0.13
Zutilo 3.9.0, 3.10.0

**Comment by qqobb** [comment]
@wobintosh: Zutilo's "Paste [...] item fields" options only show up if you have valid JSON item data on the clipboard. Try using "Copy items fields", then check if you see the "Paste [...] item fields" options.

See these discussions for more details:
https://forums.zotero.org/discussion/93399/cannot-use-paste-non-empty-item-fields-of-zutilo
https://forums.zotero.org/discussion/85593/editing-multiple-items-at-once

@AlexanderSavenkov: Is this the same issue you were experiencing?

**Comment by AlexanderSavenkov** [comment]
@qqobb, I think so. The behavior was counterintuitive, but once you figure it out, it is straightforward. Worth mentioning in the documentation.

**Comment by wobintosh** [comment]
@qqobb I was quite sure that I had valid JSON in my clipboard, but somehow it started working a few days later. Maybe it would be an option to gray out the option, if there is no (valid) JSON in clipboard?

**Comment by qqobb** [comment]
>Maybe it would be an option to gray out the option, if there is no (valid) JSON in clipboard?

@wobintosh: Yes, graying out the "Paste [...] item fields" options seems better than hiding them. We would have to check how to do that.

Note that your JSON could also be invalid due to syntax issues, see [here](https://forums.zotero.org/discussion/comment/382326/#Comment_382326).
